network-api/networkapi/wagtailpages/wagtail_hooks.py

```diff
@@ -100,3 +101,15 @@ def before_delete_page(request, page):
     if isinstance(page, ProductPage) and page.votes:
         # Delete the vote from ProductPages
         page.votes.delete()
+
+
+@hooks.register('after_edit_page')
+@hooks.register('after_delete_page')
+@hooks.register('after_publish_page')
+@hooks.register('after_unpublish_page')
+def manage_pni_cache(request, page):
+    if isinstance(page, ProductPage):
+        # Clear all of our Django-based cache.
+        # This is easier than looping through every Category x Language Code available
+        # To specifically clear PNI-based cache.
+        cache.clear()
```

Comment:
can we add/invalidate pages in a cache set? Because that would save a lot of time =D

Reply:
Also, question: does this invalidate the code we already have in place in the `products.py` file (the `invalidate_cache` function)?

Reply:
Oh look at this, this exists already! We should the invalidate_cache function in products.py and use the code in wagtail_hooks.py — only because hooks are _supposed_ to live in wagtail_hooks.py.

Reply:
Thats what I ended up doing in a new commit.

Reply:
Oh also I forgot to mention before closing this. I don't think there's a global way across all caching backends to get a cache by its name and wildcard its arguments.